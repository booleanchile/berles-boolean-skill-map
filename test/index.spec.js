import { getByLabelText, fireEvent } from '@testing-library/dom';

import { createSpiderChart } from '@app/index';

describe('Index Skills Map', () => {
  const formData = {
    'Internet': [1, 1, 1, 2, 3],
    'Backend': [1, 1, 0, 0, 0],
    'ProgramaciÃ³n': [1, 1, 3, 0, 0],
    'Control Versiones': [1, 1, 2, 3, 0],
    'HTML y A11y': [1, 1, 1, 2, 3],
    'CSS': [1, 1, 1, 2, 3],
    'Javascript': [1, 1, 3, 0, 0],
    'Testing': [1, 1, 3, 0, 0],
    'Performance': [1, 1, 0, 0, 0],
    'Devops': [1, 1, 1, 0, 0],
    'Arquitectura': [1, 3, 0, 0, 0]
  };
  let chartContainerElement;

  beforeEach(() => {
    document.body.innerHTML = `
      <svg height="800" width="800">
        <g height={800} width={800} transform="translate(400,400)"></g>
      </svg>
    `;
    chartContainerElement = document.querySelector('svg g');
  });

  it('should render skill maps on target element', () => {
    const skillsMapElement = createSpiderChart(
      chartContainerElement,
      formData,
    );
    expect(skillsMapElement).toMatchSnapshot();
  });

  it(`should execute provided callback in onClickSegment property with
    form data when click a segment`, () => {
    const onClickSegmentSpy = jest.fn();
    const skillsMapElement = createSpiderChart(
      chartContainerElement,
      formData,
      { onClickSegment: onClickSegmentSpy }
    );
    const targetIndex = 0;
    const targetSegmentLabel = Object.keys(formData)[targetIndex];
    
    fireEvent.click(
      getByLabelText(skillsMapElement, targetSegmentLabel)
    );

    expect(onClickSegmentSpy).toHaveBeenCalledWith(
      targetSegmentLabel,
      formData[targetSegmentLabel]
    );
  });
});
