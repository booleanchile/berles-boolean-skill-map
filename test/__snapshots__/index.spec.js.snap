// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Skills Map should render skill maps on target element 1`] = `
<g
  height="{800}"
  transform="translate(400,400)"
  width="{800}"
>
  <g
    aria-label="Internet"
    class="segment-1"
    role="listitem"
  >
    <path
      d="M3.489949670250108,-99.93908270190957A100,100,0,0,1,51.09521487822206,-85.96091563349192L25.54760743911103,-42.98045781674596A50,50,0,0,0,1.744974835125054,-49.969541350954785Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M5.234924505375162,-149.90862405286435A150,150,0,0,1,76.6428223173331,-128.94137345023788L51.09521487822206,-85.96091563349192A100,100,0,0,0,3.489949670250108,-99.93908270190957Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M6.979899340500216,-199.87816540381914A200,200,0,0,1,102.19042975644412,-171.92183126698384L76.6428223173331,-128.94137345023788A150,150,0,0,0,5.234924505375162,-149.90862405286435Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M8.72487417562527,-249.84770675477395A250,250,0,0,1,127.73803719555515,-214.90228908372978L102.19042975644412,-171.92183126698384A200,200,0,0,0,6.979899340500216,-199.87816540381914Z"
      fill="#2EAB6D"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M10.469849010750323,-299.8172481057287A300,300,0,0,1,153.2856446346662,-257.88274690047575L127.73803719555515,-214.90228908372978A250,250,0,0,0,8.72487417562527,-249.84770675477395Z"
      fill="#6D03D7"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M10.64434649426283,-304.8142022408242A305,305,0,0,1,155.8404053785773,-262.18079268215035A305,305,0,0,0,10.64434649426283,-304.8142022408242Z"
      fill="none"
      id="Internet"
    />
    <text>
      <textpath
        href="#Internet"
        startOffset="25%"
        style="text-anchor: middle;"
      >
        Internet
      </textpath>
    </text>
  </g>
  <g
    aria-label="Backend"
    class="segment-2"
    role="listitem"
  >
    <path
      d="M56.96707985722388,-82.1872971482861A100,100,0,0,1,89.45800972439535,-44.69076522224681L44.72900486219768,-22.345382611123405A50,50,0,0,0,28.48353992861194,-41.09364857414305Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M85.45061978583581,-123.28094572242917A150,150,0,0,1,134.18701458659305,-67.03614783337021L89.45800972439535,-44.69076522224681A100,100,0,0,0,56.96707985722388,-82.1872971482861Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M113.93415971444776,-164.3745942965722A200,200,0,0,1,178.9160194487907,-89.38153044449362L134.18701458659305,-67.03614783337021A150,150,0,0,0,85.45061978583581,-123.28094572242917Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M142.4176996430597,-205.46824287071527A250,250,0,0,1,223.6450243109884,-111.72691305561702L178.9160194487907,-89.38153044449362A200,200,0,0,0,113.93415971444776,-164.3745942965722Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M170.90123957167162,-246.56189144485833A300,300,0,0,1,268.3740291731861,-134.07229566674042L223.6450243109884,-111.72691305561702A250,250,0,0,0,142.4176996430597,-205.46824287071527Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M173.74959356453283,-250.67125630227264A305,305,0,0,1,272.8469296594059,-136.30683392785278A305,305,0,0,0,173.74959356453283,-250.67125630227264Z"
      fill="none"
      id="Backend"
    />
    <text>
      <textpath
        href="#Backend"
        startOffset="25%"
        style="text-anchor: middle;"
      >
        Backend
      </textpath>
    </text>
  </g>
  <g
    aria-label="Programaci贸n"
    class="segment-3"
    role="listitem"
  >
    <path
      d="M92.35756469968112,-38.34162545777392A100,100,0,0,1,99.41851856316546,10.76838737720388L49.70925928158273,5.38419368860194A50,50,0,0,0,46.17878234984056,-19.17081272888696Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M138.53634704952168,-57.51243818666088A150,150,0,0,1,149.1277778447482,16.15258106580582L99.41851856316546,10.76838737720388A100,100,0,0,0,92.35756469968112,-38.34162545777392Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M184.71512939936224,-76.68325091554784A200,200,0,0,1,198.83703712633093,21.53677475440776L149.1277778447482,16.15258106580582A150,150,0,0,0,138.53634704952168,-57.51243818666088Z"
      fill="#6D03D7"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M230.8939117492028,-95.8540636444348A250,250,0,0,1,248.54629640791367,26.9209684430097L198.83703712633093,21.53677475440776A200,200,0,0,0,184.71512939936224,-76.68325091554784Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M277.07269409904336,-115.02487637332176A300,300,0,0,1,298.2555556894964,32.30516213161164L248.54629640791367,26.9209684430097A250,250,0,0,0,230.8939117492028,-95.8540636444348Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M281.6905723340274,-116.94195764621045A305,305,0,0,1,303.22648161765466,32.843581500471835A305,305,0,0,0,281.6905723340274,-116.94195764621045Z"
      fill="none"
      id="Programaci贸n"
    />
    <text>
      <textpath
        href="#Programaci贸n"
        startOffset="25%"
        style="text-anchor: middle;"
      >
        Programaci贸n
      </textpath>
    </text>
  </g>
  <g
    aria-label="Control Versiones"
    class="segment-4"
    role="listitem"
  >
    <path
      d="M98.42517531735837,17.677241406601564A100,100,0,0,1,77.81435021581527,62.808653070181755L38.907175107907634,31.404326535090878A50,50,0,0,0,49.21258765867918,8.838620703300782Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M147.63776297603755,26.515862109902343A150,150,0,0,1,116.7215253237229,94.21297960527262L77.81435021581527,62.808653070181755A100,100,0,0,0,98.42517531735837,17.677241406601564Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M196.85035063471673,35.35448281320313A200,200,0,0,1,155.62870043163053,125.61730614036351L116.7215253237229,94.21297960527262A150,150,0,0,0,147.63776297603755,26.515862109902343Z"
      fill="#2EAB6D"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M246.06293829339592,44.1931035165039A250,250,0,0,1,194.53587553953815,157.02163267545438L155.62870043163053,125.61730614036351A200,200,0,0,0,196.85035063471673,35.35448281320313Z"
      fill="#6D03D7"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M295.2755259520751,53.03172421980469A300,300,0,0,1,233.4430506474458,188.42595921054524L194.53587553953815,157.02163267545438A250,250,0,0,0,246.06293829339592,44.1931035165039Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M249.00592069060883,200.9876898245816A320,320,0,0,0,314.9605610155468,56.567172501125A320,320,0,0,1,249.00592069060883,200.9876898245816Z"
      fill="none"
      id="Control Versiones"
    />
    <text>
      <textpath
        href="#Control Versiones"
        startOffset="25%"
        style="text-anchor: middle;"
      >
        Control Versiones
      </textpath>
    </text>
  </g>
  <g
    aria-label="HTML y A11y"
    class="segment-5"
    role="listitem"
  >
    <path
      d="M73.24348821083109,68.08370902580035A100,100,0,0,1,31.504675484869278,94.90761519811292L15.752337742434639,47.45380759905646A50,50,0,0,0,36.621744105415544,34.04185451290017Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M109.86523231624663,102.12556353870052A150,150,0,0,1,47.257013227303915,142.36142279716938L31.504675484869278,94.90761519811292A100,100,0,0,0,73.24348821083109,68.08370902580035Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M146.48697642166218,136.1674180516007A200,200,0,0,1,63.009350969738556,189.81523039622584L47.257013227303915,142.36142279716938A150,150,0,0,0,109.86523231624663,102.12556353870052Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M183.1087205270777,170.20927256450085A250,250,0,0,1,78.7616887121732,237.2690379952823L63.009350969738556,189.81523039622584A200,200,0,0,0,146.48697642166218,136.1674180516007Z"
      fill="#2EAB6D"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M219.73046463249327,204.25112707740104A300,300,0,0,1,94.51402645460783,284.72284559433876L78.7616887121732,237.2690379952823A250,250,0,0,0,183.1087205270777,170.20927256450085Z"
      fill="#6D03D7"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M100.81496155158169,303.7043686339614A320,320,0,0,0,234.37916227465948,217.86786888256108A320,320,0,0,1,100.81496155158169,303.7043686339614Z"
      fill="none"
      id="HTML y A11y"
    />
    <text>
      <textpath
        href="#HTML y A11y"
        startOffset="25%"
        style="text-anchor: middle;"
      >
        HTML y A11y
      </textpath>
    </text>
  </g>
  <g
    aria-label="CSS"
    class="segment-6"
    role="listitem"
  >
    <path
      d="M24.807511111122864,96.87408008580786A100,100,0,0,1,-24.807511111122853,96.87408008580786L-12.403755555561427,48.43704004290393A50,50,0,0,0,12.403755555561432,48.43704004290393Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M37.211266666684296,145.3111201287118A150,150,0,0,1,-37.21126666668428,145.3111201287118L-24.807511111122853,96.87408008580786A100,100,0,0,0,24.807511111122864,96.87408008580786Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M49.61502222224573,193.74816017161572A200,200,0,0,1,-49.615022222245706,193.74816017161572L-37.21126666668428,145.3111201287118A150,150,0,0,0,37.211266666684296,145.3111201287118Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M62.018777777807166,242.18520021451968A250,250,0,0,1,-62.01877777780714,242.18520021451968L-49.615022222245706,193.74816017161572A200,200,0,0,0,49.61502222224573,193.74816017161572Z"
      fill="#2EAB6D"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M74.42253333336859,290.6222402574236A300,300,0,0,1,-74.42253333336856,290.6222402574236L-62.01877777780714,242.18520021451968A250,250,0,0,0,62.018777777807166,242.18520021451968Z"
      fill="#6D03D7"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-79.38403555559313,309.9970562745852A320,320,0,0,0,79.38403555559317,309.9970562745852A320,320,0,0,1,-79.38403555559313,309.9970562745852Z"
      fill="none"
      id="CSS"
    />
    <text>
      <textpath
        href="#CSS"
        startOffset="25%"
        style="text-anchor: middle;"
      >
        CSS
      </textpath>
    </text>
  </g>
  <g
    aria-label="Javascript"
    class="segment-7"
    role="listitem"
  >
    <path
      d="M-31.504675484869267,94.90761519811294A100,100,0,0,1,-73.24348821083109,68.08370902580036L-36.621744105415544,34.04185451290018A50,50,0,0,0,-15.752337742434634,47.45380759905647Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-47.2570132273039,142.3614227971694A150,150,0,0,1,-109.86523231624663,102.12556353870053L-73.24348821083109,68.08370902580036A100,100,0,0,0,-31.504675484869267,94.90761519811294Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-63.009350969738534,189.81523039622587A200,200,0,0,1,-146.48697642166218,136.16741805160072L-109.86523231624663,102.12556353870053A150,150,0,0,0,-47.2570132273039,142.3614227971694Z"
      fill="#6D03D7"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-78.76168871217317,237.26903799528233A250,250,0,0,1,-183.1087205270777,170.20927256450088L-146.48697642166218,136.16741805160072A200,200,0,0,0,-63.009350969738534,189.81523039622587Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-94.5140264546078,284.7228455943388A300,300,0,0,1,-219.73046463249327,204.25112707740107L-183.1087205270777,170.20927256450088A250,250,0,0,0,-78.76168871217317,237.26903799528233Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-234.37916227465948,217.86786888256114A320,320,0,0,0,-100.81496155158165,303.7043686339614A320,320,0,0,1,-234.37916227465948,217.86786888256114Z"
      fill="none"
      id="Javascript"
    />
    <text>
      <textpath
        href="#Javascript"
        startOffset="25%"
        style="text-anchor: middle;"
      >
        Javascript
      </textpath>
    </text>
  </g>
  <g
    aria-label="Testing"
    class="segment-8"
    role="listitem"
  >
    <path
      d="M-77.81435021581525,62.808653070181755A100,100,0,0,1,-98.42517531735837,17.67724140660153L-49.21258765867918,8.838620703300766A50,50,0,0,0,-38.90717510790763,31.404326535090878Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-116.72152532372289,94.21297960527262A150,150,0,0,1,-147.63776297603755,26.515862109902297L-98.42517531735837,17.67724140660153A100,100,0,0,0,-77.81435021581525,62.808653070181755Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-155.6287004316305,125.61730614036351A200,200,0,0,1,-196.85035063471673,35.35448281320306L-147.63776297603755,26.515862109902297A150,150,0,0,0,-116.72152532372289,94.21297960527262Z"
      fill="#6D03D7"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-194.53587553953813,157.02163267545438A250,250,0,0,1,-246.06293829339592,44.19310351650383L-196.85035063471673,35.35448281320306A200,200,0,0,0,-155.6287004316305,125.61730614036351Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-233.44305064744577,188.42595921054524A300,300,0,0,1,-295.2755259520751,53.031724219804595L-246.06293829339592,44.19310351650383A250,250,0,0,0,-194.53587553953813,157.02163267545438Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-314.9605610155468,56.5671725011249A320,320,0,0,0,-249.0059206906088,200.9876898245816A320,320,0,0,1,-314.9605610155468,56.5671725011249Z"
      fill="none"
      id="Testing"
    />
    <text>
      <textpath
        href="#Testing"
        startOffset="25%"
        style="text-anchor: middle;"
      >
        Testing
      </textpath>
    </text>
  </g>
  <g
    aria-label="Performance"
    class="segment-9"
    role="listitem"
  >
    <path
      d="M-99.41851856316546,10.768387377203934A100,100,0,0,1,-92.3575646996811,-38.34162545777393L-46.17878234984055,-19.170812728886965A50,50,0,0,0,-49.70925928158273,5.384193688601967Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-149.12777784474818,16.1525810658059A150,150,0,0,1,-138.53634704952165,-57.512438186660894L-92.3575646996811,-38.34162545777393A100,100,0,0,0,-99.41851856316546,10.768387377203934Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-198.83703712633093,21.53677475440787A200,200,0,0,1,-184.7151293993622,-76.68325091554786L-138.53634704952165,-57.512438186660894A150,150,0,0,0,-149.12777784474818,16.1525810658059Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-248.54629640791364,26.920968443009837A250,250,0,0,1,-230.89391174920277,-95.85406364443483L-184.7151293993622,-76.68325091554786A200,200,0,0,0,-198.83703712633093,21.53677475440787Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-298.25555568949636,32.3051621316118A300,300,0,0,1,-277.0726940990433,-115.02487637332179L-230.89391174920277,-95.85406364443483A250,250,0,0,0,-248.54629640791364,26.920968443009837Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-303.22648161765466,32.843581500472A305,305,0,0,1,-281.69057233402737,-116.9419576462105A305,305,0,0,0,-303.22648161765466,32.843581500472Z"
      fill="none"
      id="Performance"
    />
    <text>
      <textpath
        href="#Performance"
        startOffset="25%"
        style="text-anchor: middle;"
      >
        Performance
      </textpath>
    </text>
  </g>
  <g
    aria-label="Devops"
    class="segment-10"
    role="listitem"
  >
    <path
      d="M-89.4580097243954,-44.69076522224674A100,100,0,0,1,-56.96707985722382,-82.18729714828615L-28.48353992861191,-41.093648574143074A50,50,0,0,0,-44.7290048621977,-22.34538261112337Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-134.18701458659308,-67.03614783337011A150,150,0,0,1,-85.45061978583574,-123.28094572242921L-56.96707985722382,-82.18729714828615A100,100,0,0,0,-89.4580097243954,-44.69076522224674Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-178.9160194487908,-89.38153044449348A200,200,0,0,1,-113.93415971444765,-164.3745942965723L-85.45061978583574,-123.28094572242921A150,150,0,0,0,-134.18701458659308,-67.03614783337011Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-223.64502431098848,-111.72691305561685A250,250,0,0,1,-142.41769964305956,-205.46824287071536L-113.93415971444765,-164.3745942965723A200,200,0,0,0,-178.9160194487908,-89.38153044449348Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-268.37402917318616,-134.07229566674022A300,300,0,0,1,-170.90123957167148,-246.56189144485842L-142.41769964305956,-205.46824287071536A250,250,0,0,0,-223.64502431098848,-111.72691305561685Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-272.84692965940593,-136.30683392785255A305,305,0,0,1,-173.74959356453266,-250.67125630227272A305,305,0,0,0,-272.84692965940593,-136.30683392785255Z"
      fill="none"
      id="Devops"
    />
    <text>
      <textpath
        href="#Devops"
        startOffset="25%"
        style="text-anchor: middle;"
      >
        Devops
      </textpath>
    </text>
  </g>
  <g
    aria-label="Arquitectura"
    class="segment-11"
    role="listitem"
  >
    <path
      d="M-51.09521487822207,-85.96091563349191A100,100,0,0,1,-3.489949670250076,-99.93908270190957L-1.744974835125038,-49.969541350954785A50,50,0,0,0,-25.547607439111037,-42.980457816745954Z"
      fill="#41D38C"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-76.64282231733311,-128.94137345023785A150,150,0,0,1,-5.234924505375115,-149.90862405286435L-3.489949670250076,-99.93908270190957A100,100,0,0,0,-51.09521487822207,-85.96091563349191Z"
      fill="#6D03D7"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-102.19042975644415,-171.92183126698382A200,200,0,0,1,-6.979899340500152,-199.87816540381914L-5.234924505375115,-149.90862405286435A150,150,0,0,0,-76.64282231733311,-128.94137345023785Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-127.73803719555518,-214.90228908372978A250,250,0,0,1,-8.72487417562519,-249.84770675477395L-6.979899340500152,-199.87816540381914A200,200,0,0,0,-102.19042975644415,-171.92183126698382Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-153.28564463466623,-257.8827469004757A300,300,0,0,1,-10.46984901075023,-299.8172481057287L-8.72487417562519,-249.84770675477395A250,250,0,0,0,-127.73803719555518,-214.90228908372978Z"
      fill="#C0C0C0"
      stroke="#FFF"
      stroke-width="2px"
    />
    <path
      d="M-155.84040537857732,-262.1807926821503A305,305,0,0,1,-10.644346494262733,-304.8142022408242A305,305,0,0,0,-155.84040537857732,-262.1807926821503Z"
      fill="none"
      id="Arquitectura"
    />
    <text>
      <textpath
        href="#Arquitectura"
        startOffset="25%"
        style="text-anchor: middle;"
      >
        Arquitectura
      </textpath>
    </text>
  </g>
</g>
`;
